### PR TITLE
Use m_vartype (ArrayInitializer) to get the initializer type.

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**
Fixes: https://github.com/lfortran/lfortran/issues/1065

### CODE REVIEW

File: integration_tests/arrays_20.f90 [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+    integer, allocatable          :: list_integer(:)
+    logical, allocatable          :: list_logical(:)
+    character(len=5), allocatable :: list_character(:)
+
+    list_integer   = [integer ::]
+    list_logical   = [logical ::]
+    list_character = [character(len=len(list_character)) ::]
+
+end program arrays_20
```
Comment (czgdp1807) [inline integration_tests/arrays_20.f90:12]: Please register this in `tests.toml` as well. I guess we should make sure the types are getting set correctly in ASR.

**Review by czgdp1807 — CHANGES_REQUESTED** [review]

File: integration_tests/arrays_20.f90 (hunk 1 above)
Comment (Thirumalai-Shaktivel) [reply integration_tests/arrays_20.f90:12]: Only ASR, right?

File: tests/reference/asr-arrays_20-cb6ba8f.stdout [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                            (IntegerConstant 0 (Integer 4 [])))]) 
+                            ColMajor
+                        ) 
+                        ()
+                    )
+                    (= 
+                        (Var 2 list_character) 
+                        (ArrayConstant 
+                            [] 
+                            (Character 1 1 () [((IntegerConstant 1 (Integer 4 []))
```
Comment (czgdp1807) [inline tests/reference/asr-arrays_20-cb6ba8f.stdout:83]: I guess this should have the same type as `list_character`? How is this having no length attribute. If its taking compile time value of length then it should be 5.

**Comment by certik** [comment]
If it compiles with LLVM, then compile it. But in addition, if there is some tricky stuff you want to make sure works in ASR, then add an ASR test.

File: tests/reference/asr-arrays_20-cb6ba8f.stdout (hunk 2 above)
Comment (Thirumalai-Shaktivel) [reply tests/reference/asr-arrays_20-cb6ba8f.stdout:83]: Also, can you please share how the compile time value of `len(string)` is calculated? is it using the `eval_len` in `compiletime_eval.h`?

Comment (czgdp1807) [reply tests/reference/asr-arrays_20-cb6ba8f.stdout:83]: Yes.

**Review by czgdp1807 — APPROVED** [review]
LGTM.

**Comment by Thirumalai-Shaktivel** [comment]
Thanks for the approval! Let's merge this.

**Pull request merged**